consumer/src/main/java/com/manquius/twelvefactor/client/ConsumeClientProvider.java

```diff
@@ -1,14 +1,31 @@
 package com.manquius.twelvefactor.client;
 
-public class ConsumeClientProvider {
+import java.util.ArrayList;
+import java.util.List;
+import static java.util.Optional.ofNullable;
 
-    private ConsumeClient redisConsumeClient;
+public class ConsumeClientProvider {
 
+    private ConsumeClient adapter;
 
     public ConsumeClient getClient() throws ClientCreationException {
-        if(redisConsumeClient == null) {
-            redisConsumeClient = new KafkaConsumeClient();
+        if (null == adapter) {
```

Comment:
Check these. 

https://www.baeldung.com/java-avoid-null-check
https://codeburst.io/better-null-checking-in-java-369f1701155f


Let me know if we can try some of this for null.

Reply:
This is a one time initialization, proposed changes are meaningless in this context.